src/Files.App/Converters/GridLengthToDouble.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+
+namespace Files.App.Converters;
+
+public class GridLengthToDouble : IValueConverter
+{
+	public object Convert(object value, Type targetType, object parameter, string language)
+	{
+		return value is GridLength { IsAbsolute: true } gridLength 
+			? gridLength.Value 
+			: throw new InvalidCastException("GridLength with a \"Star\"('*') or \"Auto\" value cannot be converted to double");
+	}
+
+	public object ConvertBack(object value, Type targetType, object parameter, string language)
+	{
+		return new GridLength((double)value);
+	}
+}
```

Comment:
The class should be contained inside the namespace I believe?

Reply:
I am sorry that I don't understand. Do you mean I should put this class in another namespace?

Reply:
We usually use the notation with curly brackets for the namespace scopes on File. Meaning your code should be like this

```csharp
namespace Files.App.Converters
{
    public class GridLengthToDouble : IValueConverter { ... }
}
```

Rather than: 

```csharp
namespace Files.App.Converters;

public class GridLengthToDouble : IValueConverter { ... }
```

Reply:
Ok. I'll get right on it.